app/controllers/admin/auth_controller.rb

```diff
@@ -12,9 +12,10 @@ def sign_out
 
     def callback
       admin_session = AdminSession.from_auth_hash(request.env.fetch("omniauth.auth"))
+      dfe_sign_in_user = DfeSignIn::User.find_or_create_by(dfe_sign_in_id: admin_session.user_id)
```

Comment:
This should only really happen `if admin_session.has_admin_access?`, otherwise we'll create user records even for those users that don't have access.